tests/e2e/empty_dir_test.go

```diff
@@ -0,0 +1,27 @@
+package e2e
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/pressly/goose/v3"
+	"github.com/pressly/goose/v3/internal/check"
+)
+
+func TestEmptyMigrationsDirectory(t *testing.T) {
+	t.Parallel()
+
+	migrationsDir, err := ioutil.TempDir(os.TempDir(), "migrations")
+	if err != nil {
+		check.NoError(t, err)
+	}
+	defer os.RemoveAll(migrationsDir)
+
+	_, err = newDockerDB(t)
+	check.NoError(t, err)
+	goose.SetDialect(*dialect)
+
+	_, err = goose.CollectMigrations(migrationsDir, 0, goose.MaxVersion)
+	check.Contains(t, err.Error(), "directory does not contain any kind of migration files")
```

Comment:
we shouldn't check against error string

instead, we're fine by checking `err != nil` ... or check the sentinel error with error.Is(), if we want to introduce it explicitly

Reply:
Ok, I'll introduce an error type for this.

Reply:
@pkorobeinikov maybe you can do it after we merge #362 later today

Reply:
btw: sentinel error is a value, not a type

Reply:
```
var ErrNoMigrationFilesFoundInDirectory := errors.New("...")
```

Right?

Reply:
correct, `ErrNoMigrations`

Reply:
@VojtechVitek may be i add ErrNoMigrations in my MR?

Reply:
And @pkorobeinikov add this after merge?